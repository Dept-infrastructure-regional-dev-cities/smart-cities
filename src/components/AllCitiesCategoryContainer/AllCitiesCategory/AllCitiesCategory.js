import React from 'react';
import PropTypes from 'prop-types';
import PageWrapper from '../../PageWrapper/PageWrapper';
import PageBanner from '../../PageBanner/PageBanner';
import SubCategorySummary from '../../SubCategorySummary/SubCategorySummary';
import SubCategoryCharts from '../../SubCategoryCharts/SubCategoryCharts';
import {
  INDICATORS,
  CATEGORY_IDS,
} from '../../../constants';

const AllCitiesCategory = props => (
  <PageWrapper categoryId={props.category.id}>
    <PageBanner
      colorName={props.category.colorName}
      description={props.category.description}
      indicator={props.category.heroIndicatorId}
      title={props.category.name}
      cities={props.cities}
      isContextPage={props.category.id === CATEGORY_IDS.CONTEXT}
    />

    {props.category.subCategories.map(subCategory => (
      <SubCategorySummary
        key={subCategory.name}
        {...subCategory}
        categoryId={props.category.id}
        categoryColorName={subCategory.colorName || props.category.colorName}
        cities={props.cities}
      />
    ))}

    {props.category.subCategories.map(subCategory => (
      <SubCategoryCharts
        key={subCategory.name}
        subCategory={subCategory}
        colorName={subCategory.colorName || props.category.colorName}
        cities={props.cities}
        heroIndicatorId={props.category.heroIndicatorId}
        highlightColorDark={subCategory.highlightColorDark}
      />
    ))}
  </PageWrapper>
);

AllCitiesCategory.propTypes = {
  category: PropTypes.shape({
    colorName: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    heroIndicatorId: PropTypes.oneOf(Object.keys(INDICATORS)).isRequired,
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    subCategories: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      colorName: PropTypes.string,
      charts: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string.isRequired,
        indicatorIds: PropTypes.arrayOf(
          PropTypes.oneOf(Object.keys(INDICATORS)),
        ).isRequired,
      })),
    })).isRequired,
  }).isRequired,
  cities: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    indicators: PropTypes.object.isRequired,
  })).isRequired,
};

export default AllCitiesCategory;
