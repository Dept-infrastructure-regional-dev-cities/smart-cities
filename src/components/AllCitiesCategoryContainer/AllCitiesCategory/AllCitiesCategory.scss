@import '~style/variables';

.main {
  background: $color-white;
}
