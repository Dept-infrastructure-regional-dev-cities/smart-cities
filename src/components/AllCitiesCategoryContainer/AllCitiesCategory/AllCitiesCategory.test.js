import React from 'react';
import { shallow } from 'enzyme';
import AllCitiesCategory from './AllCitiesCategory';

const defaultProps = {
  category: {
    name: 'Mock category',
    colorName: 'jobs',
    id: 'mock',
  },
  cities: [
    {
      name: 'Perth',
      indices: {}
    },
    {
      name: 'Sydney',
      indices: {}
    },
  ],
};

it('should match Snapshot', () => {
  const component = shallow(
    <AllCitiesCategory {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});
