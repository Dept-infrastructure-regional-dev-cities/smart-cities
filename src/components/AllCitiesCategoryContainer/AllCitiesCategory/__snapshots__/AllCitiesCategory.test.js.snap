// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<PageWrapper categoryId=\\"mock\\" cityId=\\"all-cities\\">
  <div className=\\"main\\">
    <h1>
      I am the &quot;Mock category&quot; page for all cities
    </h1>
    <pre>
      {
        &quot;name&quot;: &quot;Mock category&quot;,
        &quot;colorName&quot;: &quot;jobs&quot;,
        &quot;id&quot;: &quot;mock&quot;
      }
    </pre>
  </div>
</PageWrapper>"
`;
